webportal/src/app/cluster-view/services/service-info.js

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// This function will call kubernetes restful api to get node - podlist - label info, to support service view monitor page.
+
+const getServiceView = (kubeURL, namespace, callback) => {
+    $.ajax({
+        type: "GET",
+        url: kubeURL + "/api/v1/nodes",
+        dataType: "json",
+        success: function (data) {
+            var items = data.items;
+            var nodeList = []
+            var nodeInfo = []
+            for (var i in items) {
+                var node = items[i].metadata.name
+                nodeList.push(node);
+                nodeInfo.push(items[i])
+            }
+            getNodePods(kubeURL, namespace, nodeList, nodeInfo, callback)
+        }
+    });
+}
+
+const getNodePods = (kubeURL, namespace, nodeList, nodeInfo, callback) => {
+    $.ajax({
+        type: "GET",
+        url: kubeURL + "/api/v1/namespaces/" + namespace + "/pods/",
+        dataType: "json",
+        success: function (pods) {
+            var podsItems = pods.items;
+            var nodeDic = new Array();
+
+            for (var i in podsItems) {
+                var pod = podsItems[i].metadata.name
+                var nodeName = podsItems[i].spec.nodeName
+                if (nodeDic[nodeName] == null) {
+                    nodeDic[nodeName] = []
+                }
+                nodeDic[nodeName].push({ "pod": podsItems[i] })
+            }
+            var resultDic = []
+            for(var i in nodeList) {
+                resultDic.push({"node": nodeInfo[i], "podList": nodeDic[nodeList[i]]})
```

Comment:
>nodeList[i] [](start = 72, length = 11)

If you passed the whole nodeInfo in nodeList, here should be "nodeDic[nodeList[i].metadata.name".

Reply:
i pass two list , one is ip list and other is node info. use ip list as the index

Reply:
Node info list already contains ip, so why need a seperate ip list?

---
In reply to: [159605721](https://github.com/Microsoft/pai/pull/145#discussion_r159605721) [](ancestors = 159605721)